src/environments.js

```diff
@@ -1,326 +1,16 @@
 // @flow
```

Comment:
Everything except for `defineEnvironment` and the type definitions have been copied over to `array.js` while the remaining code has gone to `defineEnvironment.js`.